tendrl/monitoring_integration/graphite/__init__.py

```diff
@@ -192,127 +179,80 @@ def get_central_store_data(self, objects):
-                            {
-                                'message': "Cannot Find {0} in brick "
-                                "{1}".format(key, brick) + str(ex)
-                            }
-                        )
+                    brick_attr_value = self.resource_status_mapper(
+                        str(getattr(_brick, key))
+                    )
+                    resource_detail[key] = brick_attr_value
                 if not resource_detail == {}:
                     brick_detail.append(resource_detail)
         return brick_detail
 
-    def get_volume_details(self, objects, cluster_key):
+    def get_volume_details(self, objects, integration_id):
         volume_detail = []
-        volume_list = utils.get_resource_keys(cluster_key, "Volumes")
-        for volume in volume_list:
+        _volumes = NS.tendrl.objects.GlusterVolume(
+            integration_id=integration_id
+        ).load_all() or []
+        for _volume in _volumes:
             resource_detail = {}
-            volume_key = os.path.join(cluster_key, "Volumes", volume)
-            volume_deleted_key = os.path.join(volume_key, "deleted")
-            try:
-                is_volume_deleted = etcd_utils.read(
-                    volume_deleted_key
-                ).value
-                if is_volume_deleted.lower() == "true":
-                    continue
-            except etcd.EtcdKeyNotFound:
+            if _volume.deleted in ['true', 'True', 'TRUE']:
                 continue
             for key, value in objects["Volume"]["attrs"].items():
                 if value is None:
-                    try:
-                        attr_key = os.path.join(volume_key, key)
-                        attr_data = etcd_utils.read(attr_key)
-                        attr_value = self.resource_status_mapper(
-                            str(attr_data.value))
-                        resource_detail[key] = attr_value
-                    except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                        logger.log(
-                            "debug",
-                            NS.get("publisher_id", None),
-                            {'message': "Cannot Find {0} in volume "
-                             "{1}".format(key, volume) + str(ex)
-                             }
-                        )
+                    attr_value = self.resource_status_mapper(
+                        str(getattr(_volume, key))
+                    )
+                    resource_detail[key] = attr_value
                 else:
                     try:
-                        new_key = os.path.join(
-                            volume_key,
-                            objects["Volume"]["attrs"][
-                                key]["value"].rsplit("/", 1)[1]
-                        )
                         resp_data = self.get_object_from_central_store(
-                            new_key, objects["Volume"]["attrs"][key])
+                            key,
```

Comment:
here key is "geo_rep_session" so no object class  in this name in NS.tendrl.objects[obj_cls] it will give error.
use objects["Volume"]["attrs"][key]["value"].rsplit("/", 1)[1], i think object class name is "GeoRepSessions"

Reply:
oh ok you modified name in definition file as "GeoReplicationSession"  it will work :)